File: uri/strings/chirrin_chirrion/2583.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int C;
    if (!(cin >> C)) {
        return 0;
    }
    while (C--) {
        int N;
        cin >> N;
        unordered_set<string> owned;
        for (int i = 0; i < N; ++i) {
            string item, cmd;
            cin >> item >> cmd;
            if (cmd == "chirrin") {
                if (!owned.count(item)) {
                    owned.insert(item);
                }
            } else if (cmd == "chirrion") {
                auto it = owned.find(item);
                if (it != owned.end()) {
                    owned.erase(it);
                }
            }
        }
        cout << "TOTAL\n";
        vector<string> sorted_items(owned.begin(), owned.end());
        sort(sorted_items.begin(), sorted_items.end());
        for (const auto& item : sorted_items) {
            cout << item << '\n';
        }
    }
    return 0;
}

File: uri/strings/chirrin_chirrion/2583.rs
use std::collections::HashSet;
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    if io::stdin().read_to_string(&mut input).is_err() {
        return;
    }
    let mut iter = input.split_whitespace();
    let cases: usize = match iter.next() {
        Some(value) => value.parse().unwrap_or(0),
        None => return,
    };
    let mut outputs = Vec::new();
    for _ in 0..cases {
        let n: usize = match iter.next() {
            Some(value) => value.parse().unwrap_or(0),
            None => 0,
        };
        let mut owned: HashSet<String> = HashSet::with_capacity(n);
        for _ in 0..n {
            let item = match iter.next() {
                Some(value) => value.to_string(),
                None => String::new(),
            };
            let cmd = match iter.next() {
                Some(value) => value,
                None => continue,
            };
            if cmd == "chirrin" {
                owned.insert(item);
            } else if cmd == "chirrion" {
                owned.remove(&item);
            }
        }
        let mut items: Vec<String> = owned.into_iter().collect();
        items.sort();
        outputs.push("TOTAL".to_string());
        outputs.extend(items);
    }
    println!("{}", outputs.join("\n"));
}
